db/version_set.cc

```diff
@@ -1125,6 +1128,22 @@ VersionSet::Finalize(Version* v)
         // consider this level
         if (compact_ok)
         {
+            size_t grooming_trigger;
+
+            // which grooming trigger point?  based upon how long
+            //  since last compaction on this level
+            //   - less than 10 minutes?
+            if (micros_now<(m_CompactionStatus[level].m_LastCompaction + (1000000 * 60 *10)))
```

Comment:
I recommend pre-computing the elapsed time as "uint64_t elapsed_micros = micros_now - m_CompactionStatus[level].m_LastCompaction;" so that the comparisons become:

    uint64_t elapsed_micros = micros_now - m_CompactionStatus[level].m_LastCompaction;
    if (elapsed_micros < (1000000 * 60 *10))
        ...
    else if (elapsed_micros < (1000000 * 60 *20))